### PR TITLE
support set pf yaml config folder by cli command "pf config set a=b --path config_folder"

### PR DESCRIPTION
**Pull request by Stephen1993**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_cli/_params.py [hunk 1]
```diff
@@ -356,3 +356,13 @@ def add_param_init(parser):
         "Example: --init param1=val1 param2=val2",
         nargs="+",
     )
+
+
+def add_param_path(parser):
+    parser.add_argument(
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_params.py:362]: please also add changelog and reference doc in this PR

File: src/promptflow-devkit/promptflow/_sdk/_configuration.py [hunk 2]
```diff
@@ -247,3 +248,12 @@ def is_internal_features_enabled(self) -> Optional[bool]:
         if isinstance(result, str):
             return result.lower() == "true"
         return result is True
+
+    @classmethod
+    @contextmanager
+    def set_temp_config_path(cls, temp_path: Union[str, Path]):
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_configuration.py:254]: add test on the get config part also

File: src/promptflow-devkit/promptflow/_cli/_params.py (hunk 1 above)
Comment (Stephen1993) [reply src/promptflow-devkit/promptflow/_cli/_params.py:362]: done

File: src/promptflow-devkit/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # promptflow-devkit package
 
 ## v1.10.0 (Upcoming)
+- The `pf config set <key=value>` support set the folder where the config is saved by `--path folder` parameter,
```
Comment (wangchao1230) [inline src/promptflow-devkit/CHANGELOG.md:4]: need to update this doc: https://microsoft.github.io/promptflow/how-to-guides/set-global-configs.html

File: src/promptflow-devkit/promptflow/_cli/_params.py [hunk 4]
```diff
@@ -356,3 +356,13 @@ def add_param_init(parser):
...
+        "--path",
+        type=str,
+        required=False,
+        help="When using the set command, specify the directory of pf.yaml, "
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_cli/_params.py:366]: may refine this message

File: src/promptflow-devkit/promptflow/_sdk/_configuration.py (hunk 2 above)
Comment (Stephen1993) [reply src/promptflow-devkit/promptflow/_sdk/_configuration.py:275]: done

File: src/promptflow-devkit/promptflow/_cli/_params.py (hunk 4 above)
Comment (Stephen1993) [reply src/promptflow-devkit/promptflow/_cli/_params.py]: done

File: src/promptflow-devkit/CHANGELOG.md (hunk 3 above)
Comment (Stephen1993) [reply src/promptflow-devkit/CHANGELOG.md:4]: done

File: docs/how-to-guides/set-global-configs.md [hunk 5]
```diff
@@ -5,9 +5,13 @@ This is an experimental feature, and may change at any time. Learn [more](faq.md
 
 Promptflow supports setting global configs to avoid passing the same parameters to each command. The global configs are stored in a yaml file, which is located at `~/.promptflow/pf.yaml` by default.
 
+Additionally, promptflow supports setting configs to a specified path, and these configs will only take effect for the Promptflow program when the working directory is the specified path or its subdirectories.
```
Comment (wangchao1230) [inline docs/how-to-guides/set-global-configs.md:8]: consider change the file title to: set prompt flow configs

Comment (Stephen1993) [reply docs/how-to-guides/set-global-configs.md:8]: Okay, the name has been changed.

File: src/promptflow-devkit/promptflow/_sdk/_configuration.py [hunk 6]
```diff
@@ -51,21 +53,29 @@ class Configuration(object):
...
+
+        if self.CONFIG_PATH.is_file():
+            global_config = load_yaml(self.CONFIG_PATH)
+        else:
+            global_config = {}
+        if (cwd_config_path / file_name).is_file():
+            cwd_config = load_yaml(cwd_config_path / file_name)
+            global_config.update(cwd_config)
+
+        return global_config
```
[the remark this answers is not in the document]
Comment (Stephen1993) [reply src/promptflow-devkit/promptflow/_sdk/_configuration.py:77]: ok, I rename to all_configs.

**Review by wangchao1230 — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**